web/src/main/java/org/springframework/security/web/access/IpAddressAuthorizationManager.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.access;
+
+import java.util.function.Supplier;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.security.authorization.AuthorizationDecision;
+import org.springframework.security.authorization.AuthorizationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.util.matcher.IpAddressMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.Assert;
+
+/**
+ * An {@link AuthorizationManager} that determines if the current request is authorized
+ * based on a {@link IpAddressMatcher} evaluation.
+ *
+ * @author Evgeniy Cheban
+ */
+public class IpAddressAuthorizationManager implements AuthorizationManager<HttpServletRequest> {
```

Comment:
I'm not entirely convinced that this class should be public as these days authorizing by IP address is quite fragile. 

If necessary, I think the DSL's implementation of `hasIpAddress()` could use a private `AuthorizationManager` implementation or a lambda.